src/API/Site/Controllers/MerchantCenter/AttributeMappingController.php

```diff
@@ -0,0 +1,164 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseOptionsController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\AttributeMappingHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use WP_REST_Request as Request;
+use WP_REST_Response as Response;
+use Exception;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class for handling API requests for getting source and destination data for Attribute Mapping
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter
+ */
+class AttributeMappingController extends BaseOptionsController {
+
+	/**
+	 * @var AttributeMappingHelper
+	 */
+	private AttributeMappingHelper $attribute_mapping_helper;
+
+
+	/**
+	 * AttributeMappingController constructor.
+	 *
+	 * @param RESTServer             $server
+	 * @param AttributeMappingHelper $attribute_mapping_helper
+	 */
+	public function __construct( RESTServer $server, AttributeMappingHelper $attribute_mapping_helper ) {
+		parent::__construct( $server );
+		$this->attribute_mapping_helper = $attribute_mapping_helper;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		/**
+		 * GET the destination fields for Google Shopping
+		 */
+		$this->register_route(
+			'mc/attribute-mapping/destinations',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_attribute_mapping_destinations_read_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			],
+		);
+
+		/**
+		 * GET for getting the source data for a specific destination
+		 */
+		$this->register_route(
+			'mc/attribute-mapping/sources',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_attribute_mapping_sources_read_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
```

Comment:
If we are going to allow query arguments then we should define the schema for them with a helper function like we have done here: https://github.com/woocommerce/google-listings-and-ads/blob/2.1.1/src/API/Site/Controllers/Ads/CampaignController.php#L58

Reply:
TIL